.github/workflows/spell-checker.yml

```diff
@@ -1,25 +1,33 @@
-name: Fix typos nightly
+name: Spell Check
 
 on:
-  workflow_dispatch: {} # enable manual running of workflow.
+  pull_request:
   schedule:
-    - cron: "0 8 * * *"
+    - cron: "0 12 * * *"
+
+permissions:
+  contents: write
+  pull-requests: write
 
 jobs:
   build:
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v4
-      - uses: sobolevn/misspell-fixer-action@26173e641ba08255d0c9826b8a873f8b96f76932
+      - name: Run codespell
+        continue-on-error: true
+        run: |
+          sudo apt-get install codespell -y
+          codespell -w --skip="*.pb.go,*.pb.gw.go,*.json,*.git,*.js,*.h,*.bin,go.sum,go.mod,.github" --ignore-words=.github/.codespellignore
```

Comment:
seems to me we'd need this `.github/.codespellignore` file too :eyes:

Reply:
updated @DimitrisJim 😍

Reply:
I meant adding that file to the repo but looking at it now it seems to not have any interesting things to ignore for time being so might as well build it when the bot opens prs. Thanks!